### ISSUE TITLE
Instance Selection

### ISSUE THREAD

**Issue by tokejepsen** [pyblish/pyblish-qml#179]
**Goal**

To select the instance items in the UI.

**Motivation**

When having a lot of instances, toggling can be a pain and time consuming process and the section toggling can only get you so far. If we were able to select instance items in the UI, we could click'n'drag over the instance you want to select and toggle the instances.

**Comment by mottosso** [comment]
We'll have to find a way to separate click'n'drag for selection, and click'n'drag for scrolling. Such as alt-drag for scroll, like in Maya.

**Comment by tokejepsen** [comment]
ohh, yeah, I thought I forgot about something. I usually use the mouse wheel for scrolling.

**Comment by mottosso** [comment]
Mm, we'll have to consider the Wacom users out there as well.

**Comment by tokejepsen** [comment]
Good point. Was it out of design choices that there aren't a traditional scrollbar?

**Comment by mottosso** [comment]
Yes, minimalism at play.

**Comment by mkolar** [comment]
As lovely as such minimalism usually is, I gotta say that not a single artist I've seen use it, figured out that they can scroll by dragging. On the contrary I keep trying to select text in the terminal view as a reflex.

On the tablet note. How about using standard middle mouse type scrolling as in browser? When you press middle mouse, you can point in a direction and the website moves. That's what I use on tablet.

**Comment by BigRoy** [comment]
> As lovely as such minimalism usually is, I gotta say that not a single artist I've seen use it, figured out that they can scroll by dragging.

It took them quite a while (when not told upfront) that the list is sometimes longer than initially seen and that they can scroll by hold and dragging. Once they know it's not a problem, but they tend to miss it at first. I think it's because there's no slight hint that it's a list or scrollable element (usually noticable by a scrollbar of sorts, or something that shows what "division" you're in on the element).

> On the tablet note. How about using standard middle mouse type scrolling as in browser? When you press middle mouse, you can point in a direction and the website moves. That's what I use on tablet.

Same here. use that all the time.

**Comment by mottosso** [comment]
Ok, there's [a scrollbar implemented for the Perspective view](https://github.com/pyblish/pyblish-qml/blob/b879e2813959b7b4111f3caa9e5728163218a5c8/pyblish_qml/qml/Perspective/Viewport.qml#L117) (when you "enter" an item).

It should be easily plug-and-playable with the ListView used in the Overview. Could we open a separate issue for this, so we can properly close it once implemented?

**Comment by tokejepsen** [comment]
Done. No more sidetracking :smile:
